R/pkg/R/functions.R

```diff
@@ -2163,10 +2164,15 @@ setMethod("date_format", signature(y = "Column", x = "character"),
 #' df2 <- sql("SELECT named_struct('name', 'Bob') as people")
 #' df2 <- mutate(df2, people_json = to_json(df2$people))
 #' schema <- structType(structField("name", "string"))
-#' head(select(df2, from_json(df2$people_json, schema)))}
+#' head(select(df2, from_json(df2$people_json, schema)))
+#' head(select(df2, from_json(df2$people_json, "name STRING")))}
 #' @note from_json since 2.2.0
-setMethod("from_json", signature(x = "Column", schema = "structType"),
+setMethod("from_json", signature(x = "Column", schema = "characterOrstructType"),
           function(x, schema, as.json.array = FALSE, ...) {
+            if (is.character(schema)) {
+              schema <- structTypeFromDDL(schema)
+            }
```

Comment:
but should we maybe keep `structTypeFromDDL` here for now? I am just a bit worried that there are variants in R API alone.

I can ask this suggestion, `StructType("a STRING, b INT")`/`structType("a STRING, b INT")`, to the mailing list in coming few days if you feel okay with it.